src/taco-remote-lib/ios/iosBuild.ts

```diff
@@ -73,6 +75,50 @@ class IOSBuilder extends Builder {
         return Q({});
     }
 
+    protected collectVersionInfo(): Q.Promise<any> {
+        var self: IOSBuilder = this;
+
+        // Get the version of cordova-ios and XCode currently installed
+        let execDeferred: Q.Deferred<string> = Q.defer<string>();
+        child_process.exec("xcodebuild -version & cordova platforms version ios", { cwd: this.currentBuild.appDir } , function (error, stdout, stderr) {
+            if (error) {
+                execDeferred.reject(new Error(resources.getString("XCode83GetIosPlatformVersionWarning")));
+            } else {
+                execDeferred.resolve(stdout.toString());
+            }
+        });
+
+        return execDeferred.promise.then(function parseVersionsFromOutput(execOutput: string) {
+            // Parse XCode version from output
+            const xcodeVersionRegex = /^Xcode ((\d+)(?:\.)(\d+))?/;
+            const xcodeMatch = execOutput.match(xcodeVersionRegex);
+            if (xcodeMatch[1]){
+                self.xcodeVersion = parseFloat(xcodeMatch[1]);
+            }
+            else {
+                Logger.logWarning(resources.getString("XCodeVersionNumberFetchFailed"));
+            }
+
+            // Parse cordova-ios version from output
+            const iosVersionRegex = /ios ((\d+)(?:\.)(\d+)(?:\.)(\d+))?/;
+            const iosMatch = execOutput.match(iosVersionRegex);
+            if (iosMatch[0]) {
```

Comment:
If the match fails, then `iosMatch = null` and `iosMatch[0]` will be an error. I think you just want to check the truthiness of `iosMatch`